src/agent/README.md

```diff
@@ -125,10 +125,14 @@ The kata agent has the ability to configure agent options in guest kernel comman
 | `agent.debug_console` | Debug console flag | Allow to connect guest OS running inside hypervisor Connect using `kata-runtime exec <sandbox-id>` | boolean | `false` |
 | `agent.debug_console_vport` | Debug console port | Allow to specify the `vsock` port to connect the debugging console | integer | `0` |
 | `agent.devmode` | Developer mode | Allow the agent process to coredump | boolean | `false` |
-| `agent.hotplug_timeout` | Hotplug timeout | Allow to configure hotplug timeout(seconds) of block devices | integer | `3` |
 | `agent.guest_components_rest_api` | `api-server-rest` configuration | Select the features that the API Server Rest attestation component will run with. Valid values are `all`, `attestation`, `resource` | string | `resource` |
 | `agent.guest_components_procs` | guest-components processes | Attestation-related processes that should be spawned as children of the guest. Valid values are `none`, `attestation-agent`, `confidential-data-hub` (implies `attestation-agent`), `api-server-rest` (implies `attestation-agent` and `confidential-data-hub`) | string | `api-server-rest` |
+| `agent.hotplug_timeout` | Hotplug timeout | Allow to configure hotplug timeout(seconds) of block devices | integer | `3` |
 | `agent.https_proxy` | HTTPS proxy | Allow to configure `https_proxy` in the guest | string | `""` |
+| `agent.image_registry_auth` | Image registry credential URI | The URI to where image-rs can find the credentials for pulling images from private registries e.g. `file:///root/.docker/config.json` to read from a file in the guest image, or `kbs:///default/credentials/test` to get the file from the KBS| string | `""` |
+| `agent.enable_signature_verification` | Image security policy flag | Whether enable image security policy enforcement. If `true`, the resource indexed by URI `agent.image_policy_file` will be got to work as image pulling policy. | string | `""` |
+| `agent.image_policy_file` | Image security policy URI | The URI to where image-rs Typical policy URIs are like `file:///etc/image.json` to read from a file in the guest image, or `kbs:///default/security-policy/test` to get the file from the KBS| string | `""` |
+| `agent.simple_signing_sigstore_config` | Redhat Simple Signing scheme signature URI | The URI of the signatures that RedHat Simple Signing image signing scheme uses e.g. `file:///run/image-security/simple_signing/sigstore_config/ssample.sig` to read from a file in the guest image, or `kbs:///default/sigstore-config/testtest` to get the file from the KBS| string | `""` |
```

Comment:
Do you think these options are going to be superseded by the CDH config file? I guess maybe we could continue to support them for platforms that do not support init-data but that might be a bit messy. Is it going to be weird to add these and then later remove them?

Reply:
I'm not sure about the CDH config and init-data. My initial reaction is that we need something for platforms that aren't using it and it's also not available currently, so I'm not sure we have another approach at the moment?

Reply:
This is a compatibility issue. For platforms who support initdata we could use CDH config to support this. (see PR https://github.com/confidential-containers/guest-components/pull/608) but those who do not support initdata would still use cmdline to inject this I think. Thus I think we do not need to delete them, but add more comments to tell they are for sev/csv.

btw, in all platforms, probably only sev and csv cannot support initdata. (@huoqifeng once discussed with me about the possibility of s390x to support initdata. Sure, TDX/SNP/CCA/AzVtpm all supports)

any better suggestions?

Reply:
s390x support initdata, I got the approach described in https://github.com/confidential-containers/cloud-api-adaptor/issues/1895 worked in PR: https://github.com/confidential-containers/cloud-api-adaptor/pull/1912 and https://github.com/kata-containers/kata-containers/pull/9988, I'm preparing the PR to validate the initdata via attestation in https://github.com/confidential-containers/guest-components/pull/616 for a end to end use scenario.

Reply:
For SEV/CSV maybe we could look into the approach of sending the initdata from the KBS. SEV support in main is quite limited now, though, and I'm not totally sure what the future will be there.